### ISSUE TITLE
Numpy can not import libopenblasp-xyz.so

### ISSUE THREAD

**Issue by el-dee** [panda3d/panda3d#1358]
## Description

With latest 1.11 code, when building an app using `numpy` the generated executable crashes when `numpy` is imported as it can not find the `libopenblasp-xyz.so` although the lib is bundled.

If one call the generated executable prepended with `LD_LIBRARY_PATH=<path-to-bundle>` the application starts nicely.

## Steps to Reproduce

Add `import numpy as np` in `samples/asteroids/main.py` and modify `requirements.txt` to use a 1.11 dev wheel and add numpy as a dependency.

<pre>
Traceback (most recent call last):
  File "numpy.core", line 23, in <module>
  File "importlib._bootstrap", line 1042, in _handle_fromlist
  File "importlib._bootstrap", line 219, in _call_with_frames_removed
  File "importlib._bootstrap", line 991, in _find_and_load
  File "importlib._bootstrap", line 975, in _find_and_load_unlocked
  File "importlib._bootstrap", line 671, in _load_unlocked
  File "importlib._bootstrap", line 827, in exec_module
  File "numpy.core.multiarray", line 10, in <module>
  File "importlib._bootstrap", line 1042, in _handle_fromlist
  File "importlib._bootstrap", line 219, in _call_with_frames_removed
  File "importlib._bootstrap", line 991, in _find_and_load
  File "importlib._bootstrap", line 975, in _find_and_load_unlocked
  File "importlib._bootstrap", line 671, in _load_unlocked
  File "importlib._bootstrap", line 827, in exec_module
  File "numpy.core.overrides", line 6, in <module>
  File "importlib._bootstrap", line 991, in _find_and_load
  File "importlib._bootstrap", line 975, in _find_and_load_unlocked
  File "importlib._bootstrap", line 671, in _load_unlocked
  File "importlib._bootstrap", line 827, in exec_module
  File "numpy.core._multiarray_umath", line 1, in <module>
  File "imp", line 342, in load_dynamic
  File "importlib._bootstrap", line 702, in _load
  File "importlib._bootstrap", line 657, in _load_unlocked
  File "importlib._bootstrap", line 556, in module_from_spec
  File "importlib._bootstrap_external", line 1166, in create_module
  File "importlib._bootstrap", line 219, in _call_with_frames_removed
ImportError: libopenblas64_p-r0-742d56dc.3.20.so: cannot open shared object file: No such file or directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "__main__", line 23, in <module>
  File "importlib._bootstrap", line 991, in _find_and_load
  File "importlib._bootstrap", line 975, in _find_and_load_unlocked
  File "importlib._bootstrap", line 671, in _load_unlocked
  File "importlib._bootstrap", line 827, in exec_module
  File "numpy", line 140, in <module>
  File "importlib._bootstrap", line 1042, in _handle_fromlist
  File "importlib._bootstrap", line 219, in _call_with_frames_removed
  File "importlib._bootstrap", line 991, in _find_and_load
  File "importlib._bootstrap", line 975, in _find_and_load_unlocked
  File "importlib._bootstrap", line 671, in _load_unlocked
  File "importlib._bootstrap", line 827, in exec_module
  File "numpy.core", line 49, in <module>
ImportError: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.8 from "/home/ld/Projects/panda3d/samples/asteroids/build/linux_x86_64/asteroids"
  * The NumPy version is: "1.23.2"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: libopenblas64_p-r0-742d56dc.3.20.so: cannot open shared object file: No such file or directory
</pre>

The shared lib is present in the bundle:

<pre>
$ ls 
apt_pkg.so                                           _decimal.so                          _multibytecodec.so                 panda3d.direct.so
asteroids                                            gi._gi.so                            _multiprocessing.so                _posixshmem.so
_asyncio.so                                          _hashlib.so                          numpy.core._multiarray_tests.so    psutil._psutil_linux.so
_brotli.so                                           _json.so                             numpy.core._multiarray_umath.so    psutil._psutil_posix.so
_bz2.so                                              libgfortran-040039e1.so.5.0.0        numpy.fft._pocketfft_internal.so   PyQt5.QtCore.abi3.so
_codecs_cn.so                                        libopenblas64_p-r0-742d56dc.3.20.so  numpy.linalg.lapack_lite.so        _queue.so
_codecs_hk.so                                        libp3direct.so.1.11                  numpy.linalg._umath_linalg.so      readline.so
_codecs_iso2022.so                                   libp3dtoolconfig.so.1.11             numpy.random.bit_generator.so      resource.so
_codecs_jp.so                                        libp3dtool.so.1.11                   numpy.random._bounded_integers.so  simplejson._speedups.so
_codecs_kr.so                                        libp3interrogatedb.so.1.11           numpy.random._common.so            _ssl.so
_codecs_tw.so                                        libp3openal_audio.so                 numpy.random._generator.so         termios.so
_contextvars.so                                      libpandaexpress.so.1.11              numpy.random._mt19937.so           _testcapi.so
cryptography.hazmat.bindings._constant_time.abi3.so  libpandagl.so                        numpy.random.mtrand.so             textures
cryptography.hazmat.bindings._openssl.abi3.so        libpanda.so.1.11                     numpy.random._pcg64.so             _tkinter.so
_ctypes.so                                           libquadmath-96973f99.so.0.0.0        numpy.random._philox.so            _uuid.so
_curses.so                                           _lzma.so                             numpy.random._sfc64.so
_dbus_bindings.so                                    mmap.so                              _opcode.so
_dbus_glib_bindings.so                               models                               panda3d.core.so
</pre>

## Environment
 * Operating system: Linux/Ubuntu 20.04 and Manylinux2014
 * System architecture: x86_64
 * Panda3D version: 1.11 cf04a04e32f216517e218e2e6e298ad3f7e8f084 + fix for Py_OptimizeFlag
 * Installation method: built from source
 * Python version (if using Python): 3.7 (Manylinux) and 3.8 (Ubuntu)
 * Compiler (if using C++): N/A

**Comment by el-dee** [comment]
Side node, this worked (and still work) fine with 1.10. I did a quick diff between the branches, but nothing stood out immediately.

**Comment by el-dee** [comment]
I did a bit of digging, the main difference between the working 1.10 app and the broken 1.11 app is that in 1.10, `deploy-stub` is linked with `RPATH=$ORIGIN` while in 1.11, it is linked with `RUNPATH=$ORIGIN`.
Digging more, I realized that the 1.10 build is done using manylinux SDK and wheel while the 1.11 is build using Ubuntu 20.04. (Also after checking, the Manylinux2014 SDK and wheels are working fine too, I was confusing two setups...)

The `ld` shipped with 20.04 seems to have `--enable-new-dtags` set by default now, which mean `-rpath`option will generate a `RUNPATH` and no longer a `RPATH`

Manylinux:
<pre>
$ readelf -d cosmonium | grep PATH
 0x000000000000000f (RPATH)              Library rpath: [$ORIGIN]
</pre>

Ubuntu:
<pre>
$ readelf -d asteroids | grep PATH
 0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN]
</pre>

Rebuilding `deploy-stub` with `--disable-new-dtags` generates an app with a `RPATH` which starts up perfectly well.

The root cause for the issue is that `RUNPATH` looks similar to `RPATH` but actually isn't at all : `RPATH` does some magic to find the shared library to load, whereas `RUNPATH` is more blunt (see for more info https://github.com/spack/spack/issues/9160#issuecomment-420192189 and https://www.qt.io/blog/2011/10/28/rpath-and-runpath)

`numpy.core._multiarray_umath.so` has a `RPATH` set to `[$ORIGIN/../../numpy.libs]`, so when the loader tries to load `libopenblasp`, it first seach in the library `RPATH`, but can not find it. Following the rule, it goes back to the parent loader, but as the app has now a `RUNPATH` the loader stops instead of using it to find `libopenblasp`. With `RPATH` the loader will look in `$ORIGIN` and find the missing lib.
